### DISCUSSION TITLE
Support using `queryOptions` in `useInfiniteQuery`

### DISCUSSION THREAD

**Discussion by dora1998** [TanStack/query#7309] [Ideas] [answered]
`queryOptions` API is a good way to write queryKey and queryFn in the same place.
However, we cannot define properties for infinite query, such as `initialPageParam`, `getNextPageParam`, in `queryOptions` argument.

```typescript
// we can do this
export const contactDetailQuery = (id: string) =>
  queryOptions({
    queryKey: ['contacts', 'detail', id],
    queryFn: async () => getContact(id),
  })

// but we can't...
export const contactListQuery = (id: string) =>
  queryOptions({
    queryKey: ['contacts', 'list'],
    queryFn: async ({ pageParam }) => getContactList(pageParam),
    initialPageParam: 0,
    getNextPageParam: (lastPage, pages) => lastPage.nextCursor,
  })
```

I'm happy if `queryOptions` API also supports `useInfiniteQuery` 😄

**Comment by Semigradsky** [comment, marked as answer]
@dora1998 We have `infiniteQueryOptions` for this - https://tanstack.com/query/latest/docs/framework/react/reference/infiniteQueryOptions#infinitequeryoptions

**Reply by Semigradsky** [reply]
But not described in TS section https://tanstack.com/query/latest/docs/framework/react/typescript#typing-query-options ...

**Reply by dora1998** [reply]
Oh I missed it! Thank you 👍 
I also thought that we can join `infiniteQueryOptions` to `queryOptions` by using function overload, such as `useQuery`.

**Comment by Semigradsky** [comment]
But would be nice to have the same for `useMutation`... `mutationOptions`?
